ios/Views/PKPaymentButtonView.m

```diff
@@ -64,12 +64,16 @@ - (void)setButtonType:(NSString *) buttonType andStyle:(NSString *) buttonStyle
   PKPaymentButtonType type;
   PKPaymentButtonStyle style;
   
-  if ([buttonType isEqualToString: @"setUp"]) {
+  if ([buttonType isEqualToString: @"buy"]) {
+    type = PKPaymentButtonTypeBuy;
+  } else if ([buttonType isEqualToString: @"setUp"]) {
     type = PKPaymentButtonTypeSetUp;
-  } else if ([buttonType isEqualToString: @"continue"]) {
-    type = PKPaymentButtonTypeContinue;
   } else if ([buttonType isEqualToString: @"inStore"]) {
     type = PKPaymentButtonTypeInStore;
+  } else if ([buttonType isEqualToString: @"donate"]) {
+    type = PKPaymentButtonTypeDonate;
+  } else if ([buttonType isEqualToString: @"continue"]) {
+    type = PKPaymentButtonTypeContinue;
```

Comment:
NIT but this might be a good place for a switch case

Reply:
Good thought. I tried this in a switch statement but due to the need for the `isEqualToString` method to compare to the string, I wasn't able to get it to compile by comparing button type directly to the string literal. While this may be possible, I'm not familiar with objective c so I'm content to leave it as is since it is working.